lib/dynflow/watchers/memory_consumption_watcher.rb

```diff
@@ -0,0 +1,32 @@
+require 'get_process_mem'
+
+module Dynflow
+  module Watchers
+    class MemoryConsumptionWatcher
+
+      attr_reader :memory_limit, :world
+
+      def initialize(world, memory_limit, options)
```

Comment:
I would suggest to have an option to have a minimal limit for the world to live (we would need to set @started_at variable in the world, to be able to check this)

The reason would be to prevent immediate termination after the executor starts, without giving it a change to actually do something - this would happen if the limit would be set too low.